util/test/testutil.go

```diff
@@ -196,19 +198,72 @@ func oidcMockHandler(t *testing.T, url string) func(http.ResponseWriter, *http.R
 			require.NoError(t, err)
 			_, err = w.Write(out)
 			require.NoError(t, err)
+		case "/token":
+			if assertAzureWorkloadIdentityAssertions {
+				err := r.ParseForm()
+				require.NoError(t, err)
+
+				formData := r.Form
+				clientAssertion := formData.Get("client_assertion")
+				clientAssertionType := formData.Get("client_assertion_type")
+				assert.Equal(t, "serviceAccountToken", clientAssertion)
```

Comment:
The client assertion is not a dynamic value. The presence of the value can be asserted in the mock, but not the value itself. A test should pass a function with the necessary assertion


```go
GetAzureOIDCTestServer(t, func(w http.ResponseWriter, r *http.Request) {
  if r.RequestURI == "/token" {			
    err := r.ParseForm()
    require.NoError(t, err)
    formData := r.Form
    clientAssertion := formData.Get("client_assertion")
    assert.Equal(t, fakeTokenFileValue, clientAssertion)
  }

})

func GetAzureOIDCTestServer(t *testing.T, postHandler func(w http.ResponseWriter, r *http.Request))
....
	ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		oidcMockHandler(t, ts.URL)(w, r)
                postHandler(w, r)
	})


```

Reply:
Done, it cannot be post handler as the request is already closed by that time, Hence I defined it as a prehandler. Kept the token path flow in the common code.